tmail-backend/mailbox/encrypted/api/src/main/scala/com/linagora/tmail/encrypted/EncryptedEmailContentStore.scala

```diff
@@ -27,24 +27,28 @@ case class AttachmentNotFoundException(messageId: MessageId, position: Int) exte
 object EncryptedEmailFastView {
   def from(encryptedEmailContent: EncryptedEmailContent): EncryptedEmailFastView =
     EncryptedEmailFastView(
-      encryptedPreview = encryptedEmailContent.encryptedPreview,
+      encryptedPreview = EncryptedPreview(encryptedEmailContent.encryptedPreview),
       hasAttachment = encryptedEmailContent.hasAttachment)
 }
 
-case class EncryptedEmailFastView(encryptedPreview: String,
+case class EncryptedEmailFastView(encryptedPreview: EncryptedPreview,
                                   hasAttachment: Boolean)
 
 object EncryptedEmailDetailedView {
   def from(encryptedEmailContent: EncryptedEmailContent): EncryptedEmailDetailedView =
     EncryptedEmailDetailedView(
-      encryptedPreview = encryptedEmailContent.encryptedPreview,
-      encryptedHtml = encryptedEmailContent.encryptedHtml,
+      encryptedPreview = EncryptedPreview(encryptedEmailContent.encryptedPreview),
+      encryptedHtml = EncryptedHtml(encryptedEmailContent.encryptedHtml),
       hasAttachment = encryptedEmailContent.hasAttachment,
-      encryptedAttachmentMetadata = encryptedEmailContent.encryptedAttachmentMetadata)
+      encryptedAttachmentMetadata = EncryptedAttachmentMetadata(encryptedEmailContent.encryptedAttachmentMetadata))
 }
 
-case class EncryptedEmailDetailedView(encryptedPreview: String,
-                                      encryptedHtml: String,
+case class EncryptedAttachmentMetadata(value: Option[String]) extends AnyVal
```

Comment:
```suggestion
case class EncryptedAttachmentMetadata(value: String) extends AnyVal
```

IMO

Reply:
What happens if an email hasn't any attachments?

Reply:
ah yeah, Your mean is `Option[EncryptedAttachmentMetadata]`